src/encoded/audit/experiment.py

```diff
@@ -3920,6 +3920,59 @@ def audit_experiment_nih_institutional_certification(value, system, excluded_typ
         yield AuditFailure('missing nih_institutional_certification', detail, level='ERROR')
 
 
+def audit_experiment_eclip_queried_RNP_size_range(value, system, excluded_types):
+    '''
+    Check if libraries of eCLIP experiment and its control have matching queried_RNP_size_range.
+    '''
+    if value.get('assay_term_name') != 'eCLIP':
+        return
+
+    experiment_size_range = set()
+    control_size_range = set()
+    control_accessions = []
+
+    for rep in value['replicates']:
+        if rep.get('status') not in excluded_types and 'library' in rep:
+            lib = rep.get('library', {})
+            if 'queried_RNP_size_range' in lib:
+                experiment_size_range.add(lib['queried_RNP_size_range'])
+            else:
+                detail = ('Libraries of experiment {} are missing specification'
+                    ' of queried_RNP_size_range'.format(
+                        audit_link(path_to_text(value['@id']), value['@id'])
+                    )
+                )
+                yield AuditFailure('missing queried_RNP_size_range', detail, level='ERROR')
+
+    if 'possible_controls' in value:
+        for control in value.get('possible_controls'):
+            control_accessions.append(audit_link(path_to_text(control['@id']), control['@id']))
+            for rep in control['replicates']:
+                if rep.get('status') not in excluded_types and 'library' in rep:
+                    lib = rep.get('library', {})
+                    if 'queried_RNP_size_range' in lib:
+                        control_size_range.add(lib['queried_RNP_size_range'])
```

Comment:
I think here you can also throw "missing queried_RNAP_size_range", in the details saying it is a control that is missing it. Again here you can "return" after yield